pkg/providers/tinkerbell/assert.go

```diff
@@ -119,6 +119,20 @@ func AssertTinkerbellIPAndControlPlaneIPNotSame(spec *ClusterSpec) error {
 	return nil
 }
 
+// AssertPortsNotInUse ensures that ports 80, 42113, and 50061
+func AssertPortsNotInUse(client networkutils.NetClient) ClusterSpecAssertion {
+	return func(spec *ClusterSpec) error {
+		host := "0.0.0.0"
+		ports := []string{"80", "42113", "50061"}
+		for _, port := range ports {
+			if networkutils.IsPortInUse(client, host, port) {
+				return fmt.Errorf("port %s of host is already in use, please clean up ports", port)
+			}
+		}
```

Comment:
Instead of erroring out on the first unavailable port, can we go through the whole list and check all the ports first and then display the error with all unavailable ports?